### PR TITLE
[Mellanox][SAISERVER] Support Mellanox saiserverv1 and saiserverv2 docker

### PR DESCRIPTION
**Pull request by RyoYang**
add saiserver into buster and revert some changes

#### Why I did it
support mlnx saiserverv1 and saiserverv2
#### How I did it
Support saiserver v2 with python3 and thrift 0.14.0
add variables to support the saiserverv2
build different thrift in saithrift depends on saiserver version
build differernt versions of saiserver
make the saiserver and saiserver docker with version number
#### How to verify it
build two different versions of sasiserver in local build environment
todo: test on devices

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106

### CODE REVIEW

**Review by richardyu-ms — DISMISSED** [review]
please wait for other reviews

File: platform/mellanox/libsaithrift-dev.mk [hunk 1]
```diff
@@ -18,19 +18,28 @@
...
 SAI_VER = 0.9.4
 
-LIBSAITHRIFT_DEV = libsaithrift-dev_$(SAI_VER)_amd64.deb
+LIBSAITHRIFT_DEV = libsaithrift$(SAITHRIFT_VER)-dev_$(SAI_VER)_amd64.deb
 $(LIBSAITHRIFT_DEV)_SRC_PATH = $(SRC_PATH)/sonic-sairedis/SAI
-$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT) $(LIBTHRIFT_DEV) $(PYTHON_THRIFT) $(THRIFT_COMPILER) $(MLNX_SAI) $(MLNX_SAI_DEV)
-$(LIBSAITHRIFT_DEV)_RDEPENDS += $(LIBTHRIFT) $(MLNX_SAI)
+ifeq ($(SAITHRIFT_V2),y)
+$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT_0_13_0) $(LIBTHRIFT_DEV_0_13_0) $(PYTHON3_THRIFT_0_13_0) $(THRIFT_COMPILER_0_13_0)
+$(LIBSAITHRIFT_DEV)_RDEPENDS += $(LIBTHRIFT_0_13_0)
```
Comment (msosyak) [inline platform/mellanox/libsaithrift-dev.mk:25]: Thrift was updated to 0_14_1 in  https://github.com/Azure/sonic-buildimage/pull/9881

**Review by richardyu-ms — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Comment by liat-grozovik** [comment]
@RyoYang will you be able to have a PR against 202012 soon?

**Comment by RyoYang** [comment]
> add saiserver into buster and revert some changes
> 
> #### Why I did it
> support mlnx saiserverv1 and saiserverv2
> 
> #### How I did it
> Support saiserver v2 with python3 and thrift 0.14.0 add variables to support the saiserverv2 build different thrift in saithrift depends on saiserver version build differernt versions of saiserver make the saiserver and saiserver docker with version number



> @RyoYang will you be able to have a PR against 202012 soon?
@liat-grozovik  PR cherry-picked to 202012 branch has done, please have a look #10459
As 202012 branch's mlnx-saiserverv2 has not been verified on DUT, we may not check-in the code in the near future.

**Pull request merged**